src/Elastic.Apm/Config/AbstractConfigurationReader.cs

```diff
@@ -234,7 +237,9 @@ private bool TryParseTimeInterval(string valueAsString, out double valueInMillis
 						case TimeSuffix.S:
 							valueInMilliseconds = TimeSpan.FromSeconds(valueNoUnits).TotalMilliseconds;
 							break;
-						default: throw new ArgumentException( "Unexpected TimeSuffix value", nameof(defaultSuffix));
+						default:
+							valueInMilliseconds = -1;
+							return false;
```

Comment:
Are you sure you want to just swallow incorrect input (which most likely means a bug in agents code)? For example [TryParse(String, NumberStyles, IFormatProvider, Int32) ](https://docs.microsoft.com/en-us/dotnet/api/system.int32.tryparse?view=netframework-4.8#System_Int32_TryParse_System_String_System_Globalization_NumberStyles_System_IFormatProvider_System_Int32__)
> ArgumentException
> style is not a NumberStyles value.

Reply:
I reverted it, but if we throw an `ArgumentException` then we have to handle it - not being able to parse these values should not crash the agent even if it's an agent bug (I hope we agree at least on this).

If that's the case then we should handle it in the caller method, which is basically the same as when the method returns false - so from a single case, which is not being able to parse and returning false we made 2 cases: 1) returning false, 2) exception. Now we have to handle both, but the handling logic is the same. 

Here is the commit: https://github.com/elastic/apm-agent-dotnet/pull/374/commits/cddbba592d414cce423c4546667ee729fe0c70d9 - I think this code is very noise and repeats the same.

Nevertheless I think this is not that important - let's keep it this way.

Reply:
It's not the same - units cannot be affected by user's input that means if the units are invalid it's a bug in agent's code (or maybe some mix of assemblies from different incompatible versions, etc.) Unlike invalid configuration option value which is just an error in logging terms, the bug with invalid units is a critical event. I think in case of a critical event there's no recovery and fail-fast approach should be used which for application it usually means record the current state to help with investigation and exit as soon as possible to prevent further possibly much more disastrous business data corruption. For the agent the second part is trickier - we can consider if killing the whole application might be an option users might want to have but at the very least agent should definitely stop doing ANYTHING.

Reply:
> not being able to parse these values should not crash the agent even if it's an agent bug (I hope we agree at least on this).

Ok, we don't even agree on that one. 🤷‍♂️

I think it makes sense to not spend too much on it. I'll do whichever you want. Let me know if you want further changes.

Reply:
I think you just misread my note - I'm not talking about user's input which of course can contain invalid units, I'm talking about parameters passed by agent's code which should be valid unless there is a bug.

Reply:
Yes, I know that. I just think we overcomplicate things.

Adapting code to your suggestion.

Reply:
Regarding the way to handle - I think logging critical event is okay for now. I added the link to this discussion to https://github.com/elastic/apm-agent-dotnet/issues/208 which I think should include minimal infrastructure to handle agent's bugs (fail-fast approach or maybe something else). We can discuss it when we get to #208 (not 1.0)